doc/coq.rst

```diff
@@ -0,0 +1,507 @@
+.. _coq:
+
+***
+Coq
+***
+
+
+Dune is able to build Coq projects and plugins with support for source file
+preprocessing, extraction, and documentation.
+
+
+.. contents:: Table of Contents
+    :depth: 3
+
+.. _coq-theory:
+
+coq.theory
+----------
+
+A *Coq theory* is a collection of ``.v`` files. Together with OCaml libraries
+linking to the Coq API, a Coq theory is termed a *Coq plugin*. Both may also be
+termed a *Coq Project*. To enable Coq support in a Dune project. The language
+version should be selected in the :ref:`dune-project` file. For example:
+
+.. code:: scheme
+
+    (using coq 0.3)
+
+This will allow for the ``coq.theory`` stanza to be included in the current
+project.
+
+The basic form for defining Coq libraries is very similar to the OCaml form:
+
+.. code:: scheme
+
+    (coq.theory
+     (name <module_prefix>)
+     (package <package>)
+     (synopsis <text>)
+     (modules <ordered_set_lang>)
+     (libraries <ocaml_libraries>)
+     (flags <coq_flags>)
+     (mode <coq_native_mode>)
+     (theories <coq_theories>))
+
+The stanza will build all ``.v`` files on the given directory. The semantics of
+fields are:
+
+- ``<module_prefix>`` is a dot-separated list of valid Coq module names and
+  determines the module scope under which the theory is compiled (this
+  corresponds to Coq's ``-R`` option).
+  
+  For example, if ``<module_prefix>`` is ``foo.Bar``, the theory modules will be
+  named as ``foo.Bar.module1``, ``foo.Bar.module2``, etc. Note that modules in
+  the same theory don't see the ``foo.Bar`` prefix in the same way that OCaml
+  ``wrapped`` libraries do.
+  
+  For compatibility reasons, v1.0 of the Coq language for Dune will install a
+  theory named ``foo.Bar`` under ``foo/Bar``. Also note that Coq supports
+  composing a module path from different theories, thus you can name a theory
+  ``foo.Bar`` and a second one ``foo.Baz``, and dune will compose these
+  properly.
+
+- The ``modules`` field allows one to constrain the set of modules included in
+  the theory, similarly to its OCaml counterpart. Modules are specified in Coq
+  notation. That is to say, ``A/b.v`` is written ``A.b`` in this field.
+
+- If ``package`` is present, Dune will generate install rules for the ``.vo``
+  files of the theory. ``pkg_name`` must be a valid package name.
+  
+  Note that Coq v1.0 uses the Coq legacy install setup, where all packages share
```

Comment:
note that  `(lang coq 1.0)`  will use , I think  we should reserve that from to denote upstream Coq versions.

Reply:
It's not `(lang coq 1.0)` tho, it's `(using coq 1.0)`.